cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -2495,6 +2512,90 @@ private boolean isPodUpToDate(StatefulSet sts, Pod pod) {
             return stsGeneration == podGeneration;
         }
 
+        private final Future<ReconciliationState> getCruiseControlDescription() {
+            Promise<ReconciliationState> promise = Promise.promise();
+            vertx.createSharedWorkerExecutor("kubernetes-ops-pool").<ReconciliationState>executeBlocking(
+                future -> {
+                    try {
+                        CruiseControl cruiseControl = CruiseControl.fromCrd(kafkaAssembly, versions);
+
+                        if (cruiseControl != null) {
+                            ConfigMap logAndMetricsConfigMap = cruiseControl.generateMetricsAndLogConfigMap(
+                                    cruiseControl.getLogging() instanceof ExternalLogging ?
+                                            configMapOperations.get(kafkaAssembly.getMetadata().getNamespace(), ((ExternalLogging) cruiseControl.getLogging()).getName()) :
+                                            null);
+
+                            String configAnnotation = "";
+                            if (logAndMetricsConfigMap != null) {
+                                configAnnotation += logAndMetricsConfigMap.getData().get(ANCILLARY_CM_KEY_LOG_CONFIG);
+                            }
+                            Map<String, String> annotations = new HashMap<>();
+                            annotations.put(CruiseControl.ANNO_STRIMZI_IO_LOGGING, configAnnotation);
+
+                            this.cruiseControlMetricsAndLogsConfigMap = logAndMetricsConfigMap;
+                            this.cruiseControl = cruiseControl;
+                            this.ccDeployment = cruiseControl.generateDeployment(annotations, pfa.isOpenshift(), imagePullPolicy, imagePullSecrets);
```

Comment:
Maybe I'm missing something, but I don't see this doing any blocking operations. If that's right then there's no need for the `executeBlocking`, you can just return a completed future of `this` once you've set the fields of `this`.